### DISCUSSION TITLE
[POLL] Should we upgrade minimum PHP to 8 and Laravel 9-10?

### DISCUSSION THREAD

**Discussion by irazasyed** [irazasyed/telegram-bot-sdk#1041] [Polls]
So I've been cleaning up the code, upgrading packages, and converting all tests to pest, and so far good. 

However, it isn't compatible with PHP 8. There are some major changes with events API since I upgraded the package to v3.

I wanted to take input from the community to see if we should go ahead and drop support for PHP < 8 and Laravel < 9. Anything below these versions is anyway outdated including no security fixes.

The only problem is, we would only be able to tag the release as `v3.10` and not a major release due to the fact that, `v4` is already in the works which is another major change in itself and not compatible with v3.

The biggest advantage we would be getting is, we could support v3 for some time until v4 is actually rolled out and this would be compatible up until Laravel 10.x with all the security fixes applied, dependency packages upgraded, better tests, and bugfixes.

Thoughts? Please go ahead and participate in this poll and vote!

**Comment by irazasyed** [comment]
@jonnywilliamson @ihoru @lptn 

@ All the contributors and devs are welcome to vote.

**Reply by alies-dev** [reply]
IMO it's simple: dependency constraints should be handled by composer (both PHP and Laravel). Dropping PHP version support [is not a major change](https://sebastiandedeyne.com/composer-semver-and-underlying-dependency-changes/) - the new minor/patch package version will not be simply installed by composer, so we won't break anything.

What we need to do - just create a new branch, like `3.9.x` for cases when we need to release a new patch 3.9.x release for some reason. even more, a lot of projects started to use such version branches and even don't have master/main/dev branches.

**Reply by irazasyed** [reply]
I understand that. But there are breaking changes like with the event package upgrade from v2 to v3 ([upgrade docs](https://event.thephpleague.com/3.0/upgrade-from-2-to-3/)).

I've already pushed the `3.x` branch with the latest changes. Maybe test that in the meantime and open issues if you find any. Tests are passing but I'm yet to run and see.

**Reply by alies-dev** [reply]
oops, sorry, I wrote it in rush. I'll take a look today night or tomorrow to Events API and will share my thoughts

**Reply by irazasyed** [reply]
Great. Thanks!

**Reply by alies-dev** [reply]
> understand that. But there are breaking changes like with the event package upgrade from v2 to v3 

What is the reason of upgrading `league/event`? I assumed where some conflicts with laravel, but tries and it works with `"league/event": "^2.1"` (at least composer can resolve all dependencies).

A new major version of the package is not ready yet, 3.x version doesn't have any docs. In this situation I think it's better to have less breaking changes, especially considering quality of bug-reports in this repository (I expect to have a lot of new reports with basic conflicts that experienced developer can resolve without raising a bug here). But on other hand vote for bumping minimal PHP and Laravel versions. Optional BC breaks (like league/event) are better to avoid

**Reply by irazasyed** [reply]
So I wanted to convert the tests and fix many of the failing tests, took the Pest route and one thing led to another and various conflicts, I ended up upgrading everything, and although I've tried to avoid the breaking changes but certain things are beyond our control.

Having said that, maybe you as a developer could try upgrading to the `3.x` branch and see what fails / breaks in your existing code? That'll help us understand where we are with the BCs and accordingly see if we could make them compatible and probably make an upgrade guide.

**Reply by irazasyed** [reply]
Fixed quite a number of bugs during this upgrade. Especially with the command bus and console command.
